src/mrsimulator/utils/cartesian_tensor.py

```diff
@@ -0,0 +1,363 @@
+
+    Args:
+        euler_angles:
+            An ndarray of three Euler angles [``alpha``, ``beta``, ``gamma``]
+            using the "zyz" convention.
+        zeta_sigma:
+            Anisotropy parameter.
+        eta_sigma:
+            Asymmetry parameter
+        isotropic_component:
+            Isotropic parameter
+
+    Returns:
+        tensor: A 3x3 np.ndarray of symmetric Cartesian tensor.
+    """
+    eigenvalues = np.array(
+        [
+            -zeta_sigma * (1 + eta_sigma) / 2,
+            -zeta_sigma * (1 - eta_sigma) / 2,
+            zeta_sigma,
+        ]
+    )
+
+    # Assuming alpha_s, beta_s, gamma_s are your Euler angles
+    r = R.from_euler("zyz", euler_angles)
+    L = r.as_matrix()
+    diag_matrix = np.diag(eigenvalues)
+
+    tensor = L @ diag_matrix @ L.T
+
+    tensor += isotropic_component * np.eye(3)
+    return tensor
+
+
+def mehring_principal_components_to_maryland(
+    lambdas: List[float],
+) -> Tuple[float, float, float]:
+    """Mehring principal components to  Maryland.
+
+    Args:
+        lambdas:
+            Eigenvalues from Mehring convention.
+
+    Returns:
+        A tuple of (span, skew, isotropic) in Maryland convention where
+        ``span`` is the width, ``skew`` is the asymmetry, and ``isotropic``
+        is the isotropic components.
+    """
+    isotropic = np.mean(lambdas)
+    span = lambdas[2] - lambdas[0]
+    skew = 3 * (isotropic - lambdas[1]) / span
+    return span, skew, isotropic
+
+
+def maryland_to_mehring_principal_components(
+    isotropic: float, span: float, skew: float
+) -> np.ndarray:
+    """Maryland components to Mehring principal components.
+
+    Args:
+        span:
+            Maryland anisotropy component.
+        skew:
+            Maryland asymmetry component.
+        isotropic:
+            Maryland isotropic component.
+
+    Returns:
+        A ndarray of Mehring Eigenvalues.
+    """
+    lambda1 = isotropic - span
+    lambda2 = isotropic - skew * span / 3
+    lambda3 = (span + 2 * isotropic + skew * span / 3) / 2
+    return np.array([lambda1, lambda2, lambda3])
+
+
+def maryland_to_haeberlen_params(
+    isotropic: float, span: float, skew: float
+) -> Tuple[float, float, float]:
+    lambdas = maryland_to_mehring_principal_components(isotropic, span, skew)
+    (
+        euler_angles,
+        zeta_sigma,
+        eta_sigma,
+        isotropic_component,
+    ) = to_haeberlen_params(lambdas)
+    return zeta_sigma, eta_sigma, isotropic_component
+
+
+def haeberlen_params_to_maryland(
+    zeta_sigma: float, eta_sigma: float, isotropic_component: float
+) -> Tuple[float, float, float]:
+    tensor = from_haeberlen_params(
+        np.array([0, 0, 0]), zeta_sigma, eta_sigma, isotropic_component
+    )
+    return mehring_principal_components_to_maryland(to_mehring_params(tensor))
+
+
+def zeta_and_eta_to_xy(zeta: float, eta: float) -> Tuple[float, float]:
+    r = abs(zeta)
+    if zeta <= 0:
+        theta = np.pi / 4 * eta
```

Comment:
This funtion is already defined in `mrsimulator.models.utils.zeta_eta_to_x_y`. Removing it from here.